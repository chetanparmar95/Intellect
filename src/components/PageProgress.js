import React from 'react';
import { StyleSheet, View } from 'react-native';

const PageProgress = ({page}) => {
    return (
        <View style={styles.container}>
            <View style={[styles.progressItem, { backgroundColor: page >= 1 ? 'white' : '#ffffff55' }]} />
            <View style={[styles.progressItem, { backgroundColor: page >= 2 ? 'white' : '#ffffff55' }]} />
            <View style={[styles.progressItem, { backgroundColor: page >= 3 ? 'white' : '#ffffff55' }]} />
            <View style={[styles.progressItem, { backgroundColor: page >= 4 ? 'white' : '#ffffff55' }]} />
            <View style={[styles.progressItem, { backgroundColor: page >= 5 ? 'white' : '#ffffff55' }]} />
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row', 
        height: 4, 
        marginHorizontal: 5, 
        marginTop: 10
    },
    progressItem: {
        marginHorizontal: 5, 
        flex: 1, 
        height: 4, 
        borderRadius: 2
    }
})

export default PageProgress