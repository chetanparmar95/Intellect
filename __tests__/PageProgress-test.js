/**
 * @format
 */

import 'react-native';
import React from 'react';
import PageProgress from '../src/components/PageProgress';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders PageProgress correctly', () => {
    const tree = renderer.create(<PageProgress />).toJSON();
    expect(tree).toMatchSnapshot();
});
