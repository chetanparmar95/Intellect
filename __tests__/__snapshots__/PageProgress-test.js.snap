// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PageProgress correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 4,
      "marginHorizontal": 5,
      "marginTop": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 2,
          "flex": 1,
          "height": 4,
          "marginHorizontal": 5,
        },
        Object {
          "backgroundColor": "#ffffff55",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 2,
          "flex": 1,
          "height": 4,
          "marginHorizontal": 5,
        },
        Object {
          "backgroundColor": "#ffffff55",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 2,
          "flex": 1,
          "height": 4,
          "marginHorizontal": 5,
        },
        Object {
          "backgroundColor": "#ffffff55",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 2,
          "flex": 1,
          "height": 4,
          "marginHorizontal": 5,
        },
        Object {
          "backgroundColor": "#ffffff55",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 2,
          "flex": 1,
          "height": 4,
          "marginHorizontal": 5,
        },
        Object {
          "backgroundColor": "#ffffff55",
        },
      ]
    }
  />
</View>
`;
